src/main/java/eu/europeana/api/recommend/service/SearchApiService.java

```diff
@@ -65,22 +64,22 @@ public boolean checkRecordExists(RecordId recordId, String apikey, String token)
 
     /**
      * Given a set of recommendations, we use a reactive (non-blocking) WebClient to verify with Search API if these
-     * records still exist. Also the response of Search API is used as our final recommendation response.
+     * records still exist. Also, the response of Search API is used as our final recommendation response.
      * @param recommendations the recommendations we want to return
      * @param maxResults the maximum number of results
      * @param apikey optional, if empty apikey parameter is not included (token should be provided)
      * @param token optional, if empty the apikey parameter is used
      * @return response from Search API
      */
-    public Mono<SearchApiResponse> generateResponse(List<Recommendation> recommendations, int maxResults, String apikey, String token) {
+    public Mono<SearchApiResponse> generateResponse(List<Recommendation> recommendations, int maxResults, String apikey,String token) {
         if (recommendations == null || recommendations.isEmpty()) {
             return Mono.just(new SearchApiResponse(apikey));
         }
 
-        String query = this.generateSearchQuery(recommendations, maxResults, apikey);
+        String query = this.generateSearchQuery(recommendations, maxResults, apikey);   //Shweta: Impact for search to be verified
         Mono<SearchApiResponse> response = webClient.get()
                 .uri(query)
-                .headers(RequestUtils.generateHeaders(token))
+                .headers(RequestUtils.generateHeaders(token,null)) //Shweta: Search APi not supporting API key header - Impact to be verified
```

Comment:
better change this to apikey instead of null as you already have an implementation for SR in EA-3609. 
Or else we ill miss it